### DISCUSSION TITLE
"Please note that you need to execute this query in your SQL Server environment to get the actual data."

### DISCUSSION THREAD

**Discussion by scheung38** [FlowiseAI/Flowise#1674] [Q&A]
Connecting to SQL Server

System Message:

Based on the schema given {info} write an executable Microsoft SQL Server query for the user input, avoid MySQL query. Execute it in the database and get sql results. Make a response to user from mssql results based on the question. 
Input: "user input"
SQL query: "SQL Query here"

Human Message
{input}

Formatted Prompt Values
{
info:  "{{chat_history}}"
input: "{{question}}"
}

Prompted:

return top 5 supplier from payment_reports by reportId and amount

which AI came back with:

![Screenshot 2024-02-04 at 00 26 45](https://github.com/FlowiseAI/Flowise/assets/1273014/4564b44c-e7aa-459a-bda1-3ee76a54910f)

Then how to do the actual SQL execution and getting the result that I am expecting?
![Screenshot 2024-02-04 at 00 29 25](https://github.com/FlowiseAI/Flowise/assets/1273014/c45a9ee4-755b-45f0-a932-3049943a44d9)

![Screenshot 2024-02-04 at 00 28 53](https://github.com/FlowiseAI/Flowise/assets/1273014/4ca551ff-5611-425c-a4eb-d23437d06887)

**Comment by HenryHengZJ** [comment]
unfortunately you can only have sqldbchain OR llmchain (only one) in a chatflow

**Reply by jjamgmv** [reply]
how can connect with MSSQL with autentication?

**Reply by jjamgmv** [reply]
Can you share the structure of connection string?

**Comment by MaskedLlama04** [comment]
Greetings @scheung38  I find myself in the same situation now. I'm working with a sqlite database and my agent generates the code, however no matter what it cannot be executed. Did you manage to do it? Using maybe a custom function or something?
